crates/wash-lib/src/start/github.rs

```diff
@@ -87,3 +90,41 @@ where
         .await
         .map_or(false, |m| m.is_file())
 }
+
+/// Helper function to set up a reqwest client for performing the download
+fn get_download_client() -> reqwest::Client {
+    let proxy_username = std::env::var("WASH_PROXY_USERNAME").unwrap_or_default();
+    let proxy_password = std::env::var("WASH_PROXY_PASSWORD").unwrap_or_default();
+    // TODO(joonas): should we consider system proxy configuration here, or just explicit config?
+    let http_proxy_addr = match std::env::var("HTTP_PROXY") {
+        Ok(addr) => addr,
+        Err(_) => std::env::var("http_proxy").unwrap_or_default(),
+    };
+    let https_proxy_addr = match std::env::var("HTTPS_PROXY") {
+        Ok(addr) => addr,
+        Err(_) => std::env::var("https_proxy").unwrap_or_default(),
+    };
+
+    let mut builder = reqwest::ClientBuilder::default()
+        .user_agent(DOWNLOAD_CLIENT_USER_AGENT)
+        .with_native_certificates();
+
+    if !http_proxy_addr.is_empty() {
+        let mut proxy = reqwest::Proxy::http(http_proxy_addr).expect("failed to create http proxy");
+        if !proxy_username.is_empty() && !proxy_password.is_empty() {
+            proxy = proxy.basic_auth(&proxy_username, &proxy_password);
+        }
+        builder = builder.proxy(proxy);
```

Comment:
NIT: I think you could *probably* pull these lines out?

Reply:
So the intention is to create multiple proxies (`HTTP_PROXY` vs `HTTPS_PROXY`) while also preserving the `NO_PROXY` based on the environment variables, not just one or the other.

So the [Config inside of the ClientBuilder](https://github.com/seanmonstar/reqwest/blob/v0.12.4/src/async_impl/client.rs#L187) keeps [a `Vec<reqwest::Proxy>` of proxies](https://github.com/seanmonstar/reqwest/blob/v0.12.4/src/async_impl/client.rs#L107) that then gets passed around as necessary, which is why I am setting up a separate Proxy for each of the different schemes, because each Proxy instance can only intercept traffic for the ProxyScheme it was created for. It's also important to note that each configured Proxy should get the basic auth credentials configured for it if it's set, because we won't know which Proxy those credentials should apply to (it seems reasonable to assume that they would apply to both if set).